### PR TITLE
Fix multiple bugs with paragraph breaks

### PR DESCRIPTION
**Pull request by bterlson**
This commit fixes the following issues:

* Fragments with paragraph breaks would only emit the first paragraph. Now all paragraphs are joined with a space.
* parseText was consuming parabreaks. Text now ends at parabreak.
* Leading whitespace for a document was not consumed, leading to odd behavior with odd number of linebreaks. For example you couldn't start a list after 3 linebreaks because the linebreak after the parabreak started a non-list paragraph.

### CODE REVIEW

File: lib/emitter.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ exports.fragment = function (doc) {
     return '';
   }
 
-  return doc[0].contents.map(emit).join('');
+  return doc.reduce(function (arr, para) {
```
Comment (domenic) [inline lib/emitter.js]: Haha this is a lot of iterations and copies of the array... Might be simplest to just have a loop at this point?

**Comment by domenic** [comment]
LGTM with potential suggestion.

File: lib/emitter.js (hunk 1 above)
Comment (bterlson) [inline lib/emitter.js]: Pff, you're just a functional hater. :stuck_out_tongue:

I'm inclined to leave it as it's unlikely to ever become a bottleneck. But feel free to clean it up if you like :)

**Comment by domenic** [comment]
Published as 3.0.4!

**Pull request merged**